lib/puppet/file_serving/metadata.rb

```diff
@@ -128,9 +135,9 @@ def to_data_hash
           'type'   => checksum_type,
           'value'  => checksum
         },
-        'type'         => ftype,
-        'destination'  => destination,
-
+        'type'               => ftype,
+        'destination'        => destination,
+        'source_permissions' => source_permissions,
```

Comment:
@joshcooper Thanks for the explanation, that fix sounds kosher.

> But we likely should modify the clojure endpoint to be consistent

We don't yet have a Clojure endpoint for this, we just forward the request into Puppet via JRuby.  I suspect that upgrading our Puppet dependency will be enough for Puppet Server.

> Should I file a SERVER ticket?

Yes, please!  I don't expect much, if any, work to come out of this, but we do have some acceptance test(s) that we can clean up (un-skip) once this fix comes in so it would be good to have a ticket for it.